smt/pkg/smt/witness.go

```diff
@@ -33,7 +37,7 @@ func BuildWitness(s *SMT, rd trie.RetainDecider, ctx context.Context) (*trie.Wit
 				This algorithm adds a little bit more nodes to the witness but it ensures that all requiring nodes are included.
 			*/
 
-			retain := true
+			retain := false
```

Comment:
Should we keep it as `true`?

Reply:
I have changed it to false, because of an ineffectual assignment warning.
I think this is better, just to declare a variable, which is anyway being set a couple of lines below (https://github.com/0xPolygonHermez/cdk-erigon/pull/1363/commits/4effac3bdb3d74c20f3434ec734297432800f72a).

Reply:
Makes sense, thanks!